app/src/androidTest/java/com/github/bgabriel998/softwaredevproject/CameraPreviewTest.java

```diff
@@ -186,4 +196,83 @@ public void getCorrectOrientationString(){
         assertEquals(compassView.selectHeadingString(360), "N");
         assertEquals(compassView.selectHeadingString(1), "");
     }
+
+    /**
+     * Test that toast is displayed with correct text after taking a picture in portrait and landscape mode
+     */
+    @Test
+    public void takePictureTest() throws NoSuchMethodException, InterruptedException {
+        Context context = ApplicationProvider.getApplicationContext();
+        Assert.assertNotNull(context);
+        String pictureTakenCorrectly = context.getResources().getString(R.string.pictureTakenToast);
+
+        //Method to get the last displayed toast
+        Method getLastToast = Button1Activity.class.getDeclaredMethod("getLastToast");
+        getLastToast.setAccessible(true);
+
+        //Method to get the last displayed toast
+        Method setLastToast = Button1Activity.class.getDeclaredMethod("setLastToast", String.class);
+        getLastToast.setAccessible(true);
+
+        testRule.getScenario().onActivity(activity -> {
+            try {
+                //Check that last displayed toast is null
+                assertNull(getLastToast.invoke(activity));
+                //Set orientation to portrait
+                activity.setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE);
+                //Wait for orientation changes
+                Thread.sleep(3000);
+            } catch (InvocationTargetException | IllegalAccessException | InterruptedException e) {
+                e.printStackTrace();
+            }
+        });
+
+        //Wait for the view to be correctly displayed
+        Thread.sleep(3000);
+
+        //Take a picture, needs to be done outside of onActivity
+        onView(withId(R.id.takePicture)).perform(click());
+
+        Thread.sleep(3000);
+
+        testRule.getScenario().onActivity(activity -> {
+            try {
+                //Wait for orientation changes
+                Thread.sleep(3000);
+                //Check that correct toast was displayed
+                assertEquals(pictureTakenCorrectly, getLastToast.invoke(activity));
+                //Reset toast
+                setLastToast.invoke(activity, (Object) null);
+                //Check that last toast was reset
+                assertNull(getLastToast.invoke(activity));
+                //Rotate screen back to portrait
+                activity.setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
+                //Wait for orientation changes
+                Thread.sleep(3000);
+
+            } catch (IllegalAccessException | InvocationTargetException | InterruptedException e) {
+                e.printStackTrace();
+            }
+        });
+
+        //Wait for orientation changes
+        Thread.sleep(3000);
```

Comment:
Duplicate wait here. Consider merging the two Thread.sleep(3000) for a Thread.sleep(6000) or removing one of them

 ```suggestion
```